File: Phase01/src/Result.java
package Phase01.src;

import java.util.HashSet;

public class Result {
    private HashSet<Entry> resultSet = new HashSet<Entry>();
    private boolean isQuery;
    
    public Result(HashSet<Entry> resultSet, boolean isQuery) {
        this.resultSet = resultSet;
        this.isQuery = isQuery;
    }
    
    @Override
    public String toString() {
        StringBuilder resultStr = new StringBuilder("");
        if (this.resultSet.isEmpty()) {
            resultStr.append("No result found");
        } else {
            if (this.isQuery) {
                for(Entry entry : resultSet){
                    resultStr.append(" " + entry.getdocName());
                }
            } else {
                resultStr.append(this.resultSet.toString());
            }
        }
        return resultStr.toString();
    }
    
    public HashSet<Entry> getResultSet(){
        return this.resultSet;
    }
}

File: Phase05/app/library/Result.cs
using System.Linq;
using System.Text;

using System.Collections.Generic;
namespace library
{
    public class Result
    {
        public readonly HashSet<Entry> ResultSet;
        public readonly bool IsQuery;

        public Result(HashSet<Entry> ResultSet, bool IsQuery)
        {
            this.ResultSet = ResultSet;
            this.IsQuery = IsQuery;
        }

        public override string ToString()
        {
            var resultStr = new StringBuilder("");
            if (this.ResultSet.Count == 0)
                resultStr.Append("No result found");
            else
            {
                if (this.IsQuery)
                {
                    foreach (Entry entry in ResultSet)
                        resultStr.Append(" " + entry.DocName);
                }
                else
                    resultStr.Append(string.Join(", ", this.ResultSet));
            }
            return resultStr.ToString();
        }
    }
}